woocommerce/woocommerce
# Enhance getInputProps to allow passing of non-overridden props

### All Submissions:






### Changes proposed in this Pull Request:

Previously, adding an `onChange` handler or anything else that collided with `getInputProps` would require recalling `getInputProps.someFunction` inside the new handler.  This PR introduces a second argument to `getInputProps` which passes props to the control without overriding the original handlers where possible.

Some props will not override the original handler (e.g., if you pass on `onChange` it will be combined with the `Form` component's `onChange`) and some will override (e.g., `value` can not be combined and the consumer prop will override the original value).

This PR also uncovered numerous TS issues, so a `getCheckboxProps` was introduced into the `Form` component to omit invalid props (e.g., `value` for checkboxes).  More specific prop getters are probably in order (`getSelectProps`), but I think we can iterate on these as we go.

Finally, a `sanitize` method was added as part of the optional consumer input props.  This is a shortcut to sanitizing the input value and handling changes on blur.

**TL;DR:**

* `getInputProps` now takes a second argument to combine/override props where necessary
* `sanitize` method can be added to an input to handle sanitization on blur
* `getCheckboxProps` was introduced for props specific to checkboxes

Closes #34996 .

### How to test the changes in this Pull Request:

1. Navigate to the new product page `Products -> Add new (MVP)`
2. Check that styling for checkboxes is still correct
3. Add values to the price and dimension fields, using thousand commas and decimal separators (depending your locale)
4. Make sure they are sanitized on blur and persist correctly on save

### Other information:




### FOR PR REVIEWER ONLY:
